### PR TITLE
upgrade proto to 3.18.0 && grpc to 1.42.1

### LINKED ISSUE #907: Make GraphScope support on 'Apple Silicon CPU'
**Is your feature request related to a problem? Please describe.**

```shell
$ uname -a
arm64

$ pip3 install graphscope
Error: is not a supported wheel on this platform
```

In addition, it has been observed that protobuf are impacted by the `Apple CPU`. Specifically, the following packages start to support `M1` chip from certain versions:
```
<properties>
        <protobuf.version>3.18.0</protobuf.version>   <!-- now it is 3.12.0 -->
        <protoc.grpc.version>1.42.1</protoc.grpc.version>  <!-- now it is 1.35.0 -->
        <grpc.version>1.42.1</grpc.version>  <!-- now it is 1.35.0 -->
    </properties>
```

- [ ] Support compile and run GraphScope on `Apple CPU`
- [ ] Release pip package
- [ ] Protobuf refactor.

### PR DESCRIPTION
**Pull request by shirly121**
## What do these changes do?
upgrade proto to 3.18.0 && grpc to 1.42.1 of java-end under gaia and gie

## Related issue number

Fixes #907

### CODE REVIEW

**Comment by shirly121** [comment]
There remains some compiling errors of data-load-tools to be solved:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.2.1:shade (default) on project data_load_tools: Error creating shaded jar: Problem shading JAR /Users/zhouxiaoli/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.67/bcprov-jdk15on-1.67.jar entry META-INF/versions/15/org/bouncycastle/jcajce/provider/asymmetric/edec/SignatureSpi$EdDSA.class: java.lang.IllegalArgumentException: Unsupported class file major version 59 -> [Help 1]

File: interactive_engine/pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <log4j2.version>[2.6.1,)</log4j2.version>
     <hadoop.version>2.8.4</hadoop.version>
-    <grpc.version>1.30.1</grpc.version>
+    <grpc.version>1.42.1</grpc.version>
     <protobuf.version>3.5.1</protobuf.version>
```
Comment (yecol) [inline interactive_engine/pom.xml:41]: remove this line?

**Comment by lidongze0629** [comment]
@tianliplus Could you have time to look at this pr and upgrade the `grpc/proto` version to make the CI passed? Thanks.

**Comment by tianliplus** [comment]
> @tianliplus Could you have time to look at this pr and upgrade the `grpc/proto` version to make the CI passed? Thanks.

The error shoule be caused by out-dated code. This pr compile passed on the most recent code.

**Comment by longbinlai** [comment]
Can this be merged? @lidongze0629 @yecol

**Review by sighingnow — APPROVED** [review]

**Comment by sighingnow** [comment]
> Can this be merged? @lidongze0629 @yecol

Merged.

**Pull request merged**